source/Ship.cpp

```diff
@@ -1825,13 +1885,39 @@ void Ship::Move(vector<Visual> &visuals, list<shared_ptr<Flotsam>> &flotsam)
 		if(applyAfterburner)
 		{
 			thrust = attributes.Get("afterburner thrust");
-			double fuelCost = attributes.Get("afterburner fuel");
+			double shieldCost = attributes.Get("afterburner shields");
+			double hullCost = attributes.Get("afterburner hull");
 			double energyCost = attributes.Get("afterburner energy");
-			if(thrust && fuel >= fuelCost && energy >= energyCost)
+			double fuelCost = attributes.Get("afterburner fuel");
+			double heatCost = -attributes.Get("afterburner heat");
+
+			double dischargeCost = attributes.Get("afterburner discharge");
+			double corrosionCost = attributes.Get("afterburner corrosion");
+			double ionCost = attributes.Get("afterburner ion");
+			double leakageCost = attributes.Get("afterburner leakage");
+			double burningCost = attributes.Get("afterburner burning");
+
+			double slownessCost = attributes.Get("afterburner slowing");
+			double disruptionCost = attributes.Get("afterburner disruption");
+
+			if(thrust && shields >= shieldCost && hull >= hullCost
+				&& energy >= energyCost && fuel >= fuelCost && heat >= heatCost)
```

Comment:
I'm pretty sure that checking the heat is incorrect here.

Reply:
heatCost is the negative of the afterburner heat. If afterburner heat is positive, since heat never drops below 0, heat will always be greater than heatCost. But if afterburner heat is negative, heatCost becomes positive and gets used as a resource for the afterburner, and not having enough heat blocks use.